### DISCUSSION TITLE
Quality assessment of the obtained time-series and velocities

### DISCUSSION THREAD

**Discussion by miproietto** [gmtsar/gmtsar#983] [Q&A]
Dear GMTSAR developers,

I am currently processing InSAR data as part of my PhD research to estimate terrestrial water storage (TWS) over the Po Valley in Italy. My starting point is this paper: https://doi.org/10.1029/2023EA003326, and my objective is to improve spatial resolution. We have decided on a final pixel resolution of 100 meters for this regional study. To achieve this, I plan to process several frames over the Po Valley (~20 frames), starting with Path 117, Frame 143, ascending.

I closely followed the steps outlined in this tutorial: https://topex.ucsd.edu/gmtsar/tar/sentinel_time_series_5.pdf. Additionally, before performing the inversion, I pinned the interferograms to a point where the GNSS station BOLG_GPS is located by subtracting the phase value at this position at every point on the grid for each unwrapped interferogram. I then ran the inversion algorithm multiple times to fine-tune the smooth and atm parameters. Based on the quality indicators I used, the combination of a smooth value of 7 and atm value of 1 yielded the best results.

Furthermore, I calibrated the interferograms using GNSS displacement data and then performed the inversion with the same parameters (smooth = 7, atm = 1). The calibration was done in two ways:
1. Using the script provided by the GMTSAR suite (_correct_insar_with_gnss.csh_).
2. Using a custom script I wrote, which fits a plane to the displacement differences between GNSS and SAR, taking the average SAR displacement within a 25-pixel-wide box centered on each station location.  I followed this paper: http://dx.doi.org/10.1109/TGRS.2019.2934118.

  
I then compared these three runs (no_calibrated, calibrated_gmtsar, and calibrated_plane) with the 62 GNSS stations available within the frame. The comparison was based on the following quality indicators:
1.  Displacement:
        ◦ Root mean square error (RMSE) between SAR and GNSS, resampled to 12 days.
        ◦ Kendall tau correlation between de-trended and resampled (12 days) GNSS and SAR displacement time series.
2. Velocity:
        ◦ Velocity differences analysis, with a histogram showing estimates of the mean and standard deviation.

One important note: the reference point selected is deforming at a rate of -2.8 mm/yr (in LOS) according to the GNSS station. The displacement of the reference point was introduced in the InSAR data by resampling the GNSS displacement in LOS to 12 days and adding it to each point on the grids.

I Attached a zip folder (discussion_gmtsar) containing the results:
discussion_gmtsar/
├── displacement
│       ├── correlation_cdf-calibrated_gmtsar.pdf  
│       ├── correlation_cdf-calibrated_plane.pdf  
│       ├── correlation_cdf-no_calibrated.pdf
│       ├── correlations_distance-calibrated_gmtsar.pdf
│       ├── correlations_distance-calibrated_plane.pdf
│       ├── correlations_distance-no_calibrated.pdf
│       ├── dispcomp_data_STATION_NAME.pdf 
│       ├── rmse_cdf-calibrated_gmtsar.pdf
│       ├── rmse_cdf-calibrated_plane.pdf
│       ├── rmse_cdf-no_calibrated.pdf
│       ├── rmse_distance-calibrated_gmtsar.pdf
│       ├── rmse_distance-calibrated_plane.pdf
│       ├── rmse_distance-no_calibrated.pdf
│       └── script.log
├── general_info
│       ├── avg_coherence.png
│       ├── network.pdf
│       ├── parameters_processing
│       ├── unwrap_*.png
│       └── wrap_*.png
└── velocity
          ├── sar_vel_calibrated_gmtsar.pdf
          ├── sar_vel_calibrated_plane.pdf
          ├── sar_vel_no_calibrated.pdf
          ├── script.log
          ├── vel_diff_histogram-calibrated_gmtsar.pdf
          ├── vel_diff_histogram-calibrated_plane.pdf
          ├── vel_diff_histogram-no_calibrated.pdf
          ├── vel_scatter_distance-calibrated_gmtsar.pdf
          ├── vel_scatter_distance-calibrated_plane.pdf
          └── vel_scatter_distance-no_calibrated.pdf

**Legend of the products:**
1.  Naming conventions:
        ◦ no_calibrated: Run not calibrated with GNSS.
        ◦ calibrated_gmtsar: Run calibrated with GNSS using correct_insar_with_gnss.csh.
        ◦ calibrated_plane: Run calibrated with GNSS using the planar algorithm.
 2.    Displacement:
        ◦ correlation_cdf-*: Cumulative distribution functions of correlation coefficients.
        ◦ correlations_distance-*: Correlation coefficients as a function of distance from the reference point.
        ◦ dispcomp_data_*: Point-by-point comparison between the three SAR runs and the 63 GNSS stations.
        ◦ rmse_cdf-*: Cumulative distribution functions of RMSE.
        ◦ rmse_distance-*: RMSE as a function of distance from the reference point.
3. Velocity:
        ◦ sar_vel_*: Plot of the velocity grids with the GNSS stations' positions.
        ◦ vel_diff_histogram-*: Histogram of velocity differences between GNSS and SAR.
        ◦ vel_scatter_distance-*: Scatter plot of velocity differences between GNSS and SAR, with distance from the reference point as a colour palette.
4. General Info:
        ◦ avg_coherence.png: Average coherence across the frame.
        ◦ network.pdf: Plot of the SBAS network.
        ◦ parameters_processing: File containing all processing parameters used.
        ◦ unwrap_*.png:Two unwrapped interferograms.
        ◦  wrap_*.png: Two wrapped interferograms.

I have a doubt about my results: Why do the InSAR time series show different offsets at different points on the grid? i.e., why do the hypothetical intercepts of the linear regressions of these time series vary from point to point? Is this normal? This issue is particularly noticeable when comparing _dispcomp_data_PADO_GPS.pdf_ to _dispcomp_data_TREX_GPS.pdf_, for example. Consequently, the RMSE indicator might be biased, as it reflects these offsets, considering that the intercept of the GNSS time series is set to zero at the first SAR date.

What do you think about these results? Are they acceptable?, or can they be improved in some way? Did I make any mistakes during processing?

If anything I wrote is unclear, please let me know, and I will try to clarify it further.

Thank you for your attention and patience.
Best regards,
Michele Proietto

================================================================================================
OS: CentOS Linux release 7.9.2009
GMTSAR version: 6.2
GMT version: 6.5.0
pygmt version: 0.11.0

[discussion_gmtsar.zip](https://github.com/user-attachments/files/16762719/discussion_gmtsar.zip)

**Comment by Xiaohua-Eric-Xu** [comment]
Hi Michele,

There are several aspect that may cause offset between InSAR and GNSS differences even after integration.
1) The InSAR time-series are not well corrected for the first a few and last a few dates due to the one-sided constraints when producing the time-series. This is particularly true if you are using the -atm option for common-scene stacking mitigation on atmospheric noise. Usually when making a comparison, the mean or median difference is removed. This should be easy to understand due to the spread of noise, similar case could happen even if you tie the first data point for two randomly produced time-series (with noise). 
2) The integration script only trusts GNSS over a specific wavelength, so it will only try to make them match over the large scale and will allow differences exist between InSAR and GNSS, especially when a longer wavelength is picked for integration.
3) InSAR and GNSS are essentially measuring different things, as the deeper coupling depth for GNSS makes it insensitive to the shallower layers of soil while InSAR is sensitive to this. So fi there are different seasonal components, this is normal. (see attached)
4) choices on the smooth parameter may also cause differences.
5) sometimes the measurement from InSAR can be biased by decorrelation and will be much noisier than GNSS. 

Overall, it seems that you are looking at a not so active area where lots of GNSS stations have flat rates. Despite that, the matching between GNSS and InSAR is already pretty good I would say, such as BOLO, BOLG, and BLGN

![image](https://github.com/user-attachments/assets/0bbdf076-5d80-434b-a13f-561af2168cb9)
